plaso/engine/artifact_filters.py

```diff
@@ -0,0 +1,239 @@
+# -*- coding: utf-8 -*-
+"""Artifact filters processing."""
+
+from __future__ import unicode_literals
+
+import logging
+import re
+
+from artifacts import definitions as artifact_types
+
+from dfvfs.helpers import file_system_searcher
+from dfwinreg import registry_searcher
+
+from plaso.engine import path_helper
+
+
+ARTIFACT_FILTERS = 'ARTIFACT_FILTERS'
+COMPATIBLE_DFWINREG_KEYS = ['HKEY_LOCAL_MACHINE',
+                            'HKEY_LOCAL_MACHINE\\SYSTEM',
+                            'HKEY_LOCAL_MACHINE\\SOFTWARE',
+                            'HKEY_LOCAL_MACHINE\\SAM',
+                            'HKEY_LOCAL_MACHINE\\SECURITY']
+STANDARD_OS_FILTERS = ['windows',
+                       'linux',
+                       'darwin']
+RECURSIVE_GLOB_LIMIT = 10
+
+
+class ArtifactFilters(object):
+  """Forensic artifacts filters.
+
+  Build extraction and parsing filters from forensic artifact definitions.
+
+  Forensic artifacts are defined in:
+  https://github.com/ForensicArtifacts/artifacts/blob/master/docs/Artifacts%20definition%20format%20and%20style%20guide.asciidoc
+  """
+
+  def __init__(self, artifacts_registry, filters, knowledge_base):
+    """Initializes a filter file.
+
+    Args:
+      path (str): path to a file that contains one or more forensic artifacts.
+      knowledge_base (KnowledgeBase): Knowledge base for Log2Timeline.
+    """
+    super(ArtifactFilters, self).__init__()
+    self._artifacts_registry = artifacts_registry
+    self._filters = filters
+    self._knowledge_base = knowledge_base
+
+  def BuildFindSpecs(self, environment_variables=None):
+    """Build find specification from a forensic artifacts file.
```

Comment:
"from a forensic artifacts file." => "from artifact definitions."

Reply:
Merged in #1883